sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala

```diff
@@ -353,7 +353,7 @@ class StateStoreChangelogReaderFactory(
   }
   protected val input: DataInputStream = decompressStream(sourceStream)
 
-  private def readVersion(): Short = {
+  private lazy val readVersion: Short = {
```

Comment:
nit: it's no longer a method but a value; so probably `version` is the better name?

Reply:
I renamed it `changeLogVersion`

Reply:
I feel like the part "changeLog" is redundant because this only deals with changelog, but not a huge deal.

Reply:
the current naming leaves no room for ambiguity.